inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_mul_add_to_scaleshift_or_power.cpp

```diff
@@ -124,20 +129,23 @@ void ngraph::pass::ConvertMulAddToScaleShiftOrPower::convert_mul_add_to_scaleshi
             }
         }
 
-        auto res1 = check_constant(const_weights_node, data_node.get_shape());
-        auto res2 = check_constant(const_bias_node, mul_node->get_output_shape(0));
+        auto res1 = check_constant(const_weights_node, data_node.get_partial_shape());
+        auto res2 = check_constant(const_bias_node, mul_node->get_output_partial_shape(0));
+
+        const auto output_shape = add_node->get_output_partial_shape(0);
+        const auto output_shape_rank = output_shape.rank().get_length();
 
         if (res1 == CONVERSION_RESULT::NONE || res2 == CONVERSION_RESULT::NONE ||
-            ((res1 == CONVERSION_RESULT::SCALE_SHIFT || res2 == CONVERSION_RESULT::SCALE_SHIFT) && add_node->get_shape().size() < 4)) {
+            ((res1 == CONVERSION_RESULT::SCALE_SHIFT || res2 == CONVERSION_RESULT::SCALE_SHIFT) && output_shape_rank < 4)) {
```

Comment:
I think that it is better to check that the output shape rank is exactly 4 (not more or equal) inside the `check_constant` function and return `CONVERSION_RESULT::SCALE_SHIFT` value only if it is true. If we do so then there is no need in line 139 of this file and it looks more correct from logical point of view.

Reply:
check_constant is a generic function and shouldn't know anything about plugin constraints. This particular statement was needed by clDNN as I remember but MKLDNN supports not only 4D cases. But now we have mechanism to ask plugins which ops do they support and this statement will be replaced in future. So I prefer to keep check_constant logic as it was before.